azurelinuxagent/common/osutil/default.py

```diff
@@ -1192,26 +1192,28 @@ def get_mount_point(self, mountlist, device):
                     return tokens[2] if len(tokens) > 2 else None
         return None
 
-    def device_for_ide_port(self, port_id):
+    @staticmethod
+    def search_for_resource_disk(gen1_device_prefix, gen2_device_id):
         """
-        Return device name attached to ide port 'n'.
+        Search the filesystem for a device by ID or prefix.
+
+        Args:
+            gen1_device_prefix (str): Gen1 resource disk prefix.
+            gen2_device_id (str): Gen2 resource device ID.
+
+        Returns:
+            str: The found device.
         """
-        if port_id > 3:
-            return None
-        g0 = "00000000"
-        if port_id > 1:
-            g0 = "00000001"
-            port_id = port_id - 2
+
         device = None
         path = "/sys/bus/vmbus/devices/"
         # We have to try device IDs for both Gen1 and Gen2 VMs.
-        gen1_device_id_prefix = '{0}-000{1}'.format(g0, port_id)
         if os.path.exists(path):
             try:
                 for vmbus in os.listdir(path):
                     deviceid = fileutil.read_file(os.path.join(path, vmbus, "device_id"))
                     guid = deviceid.strip('{}')
```

Comment:
In my testing, there were newlines at the end of the data returned from read_file; do you need to add `\n` to the `strip()` call?